docs/notebooks/LORIS_API_Part2_Python-script.ipynb

```diff
@@ -0,0 +1,2159 @@
+    },
+    "colab_type": "code",
+    "id": "VeCuzjwbZ_HG",
+    "outputId": "3c13b4b1-4de5-452e-ff2c-98c26a312bf4",
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "payload = {\n",
+    "    'username': input('username: '), \n",
+    "    'password': getpass.getpass('password: ')\n",
+    "}\n",
+    "\n",
+    "response = requests.post(\n",
+    "    url = '/'.join([baseurl, 'login']),\n",
+    "    json = payload,\n",
+    "    verify = False\n",
+    ")\n",
+    "\n",
+    "text = response.content.decode('ascii')\n",
+    "\n",
+    "data = json.loads(text)\n",
+    "\n",
+    "prettyPrint(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "5D9tnYTTZ_HM"
+   },
+   "source": [
+    "*Store the token in a variable for later*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "NGatwtA2Z_HN",
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "token = data['token']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "S6G0Qn6uSyBn"
+   },
+   "source": [
+    "The requests thoughout this tutorial are very similar, only the URL changes. We will define 5 functions that will make the tutorial easier to read.\n",
+    "\n",
+    "For more information on HTTP requests and their differences, you can refer to https://github.com/aces/Loris/blob/main/modules/api/docs/LorisRESTAPI_v0.0.3.md#10-overview"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "unsItgX_SxPF"
+   },
+   "outputs": [],
+   "source": [
+    "def GET(url):\n",
+    "    \"\"\"\n",
+    "    Function to send an HTTP GET request\n",
+    "\n",
+    "    Parameters: baseurl (URL of the api, e.g. https://demo.loris.ca/api/v0.0.3)\n",
+    "\n",
+    "    Returns: projects (Information on all projects)\n",
+    "    \"\"\"\n",
+    "    response = json.loads(requests.get(\n",
+    "        url = url,\n",
+    "        verify = False,\n",
+    "        headers = {'Authorization': 'Bearer %s' % token}\n",
+    "    ).content.decode('ascii'))\n",
+    "    return response\n",
+    "\n",
+    "def GETFile(url):\n",
+    "    \"\"\"\n",
+    "    Function to send an HTTP GET request to download files\n",
+    "\n",
+    "    Parameters: baseurl (URL of the api, e.g. https://demo.loris.ca/api/v0.0.3)\n",
+    "\n",
+    "    Returns: projects (Information on all projects)\n",
+    "    \"\"\"\n",
+    "    filename = file['Filename']\n",
+    "    image = requests.get(\n",
+    "        url = url,\n",
+    "        verify = False,\n",
+    "        headers = {'Authorization': 'Bearer %s' % token}\n",
+    "    )\n",
+    "    f = open(filename, \"w+b\")\n",
+    "    f.write(bytes(image.content))\n",
+    "\n",
+    "def PUT(url, json_input):\n",
```

Comment:
Those functions are a nice addition but I think they should return the response instead of printing. There are information in those responses (body, headers) that would be useful to the users.

Reply:
I changed it to return the response. I think you are right, it is more convenient to return the response.